### PR TITLE
Evasion and wheelchair bound fix

### PR DESCRIPTION
**Pull request by Aviu00**
## About the PR

Fixed evasion mispredicts.
Fixed wheelchair bound people baing able to walk.
Now you can't stand up if any of your legs are broken, also legs and arms can't evade now. I want this for armblade and esword delimbing combat.

## Technical details

Random isn't predicted.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl: Aviu
- fix: Wheelchair bound people can no longer stand up.
- tweak: Only head can evade now.
- tweak: People with at least 1 broken leg can't stand up.
- fix: Fix evasion popups appearing when they shouldn't.
- fix: Mobs always target torso now, so the person being hit by them can't evade.

### CODE REVIEW

File: Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs [hunk 1]
```diff
@@ -104,11 +104,15 @@ private void OnStateEnteredSubscribers(EntityUid target, MobStateComponent compo
                 _appearance.SetData(target, MobStateVisuals.State, MobState.Alive);
                 break;
             case MobState.Critical:
+                if (_net.IsClient) // Shitmed - Evasion breaks stuff
+                    break;
```
Comment (gluesniffler) [inline Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs:108]: Kill this, evasion is now predicted.

File: Content.Shared/Body/Systems/SharedBodySystem.Parts.cs [hunk 2]
```diff
@@ -371,6 +371,10 @@ private void AddLeg(Entity<BodyPartComponent> legEnt, Entity<BodyComponent?> bod
 
     private void RemoveLeg(Entity<BodyPartComponent> legEnt, Entity<BodyComponent?> bodyEnt)
     {
+        // If evasion procs on server side, the entity will still fall down on client
+        if (_net.IsClient)
+            return;
+
```
Comment (gluesniffler) [inline Content.Shared/Body/Systems/SharedBodySystem.Parts.cs:377]:
```suggestion

File: Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs (hunk 1 above)
Comment (gluesniffler) [inline Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs:108]:
```suggestion

File: Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs [hunk 3]
```diff
@@ -104,11 +104,15 @@ private void OnStateEnteredSubscribers(EntityUid target, MobStateComponent compo
...
                 _standing.Down(target);
                 RaiseLocalEvent(target, new DropHandItemsEvent()); // Goobstation
                 _appearance.SetData(target, MobStateVisuals.State, MobState.Critical);
                 break;
             case MobState.Dead:
+                if (_net.IsClient) // Shitmed - Evasion breaks stuff
+                    break;
```
Comment (gluesniffler) [inline Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs:115]:
```suggestion

File: Content.Shared/Mobs/Systems/MobStateSystem.cs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 using Content.Shared.Administration.Logs;
 using Content.Shared.Mobs.Components;
 using Content.Shared.Standing;
+using Robust.Shared.Network;
```
Comment (gluesniffler) [inline Content.Shared/Mobs/Systems/MobStateSystem.cs:5]:
```suggestion

File: Content.Shared/Mobs/Systems/MobStateSystem.cs [hunk 5]
```diff
@@ -16,6 +17,7 @@ public partial class MobStateSystem : EntitySystem
     [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
     [Dependency] private readonly ILogManager _logManager = default!;
     [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly INetManager _net = default!; // Shitmed
```
Comment (gluesniffler) [inline Content.Shared/Mobs/Systems/MobStateSystem.cs:20]:
```suggestion

**Review by gluesniffler — CHANGES_REQUESTED** [review]
Okay so the way you were handling evasion is a bit flawed, because even if we do prevent that stuff from happening, you still get mispredicts both visually and in the medhud, since DamageableSystem is predicted, and thus the client runs without an evasion happening.

I'll send you on discord my tweaked SharedBodySystem.Targeting so you can just paste it in this PR. Biggest change is just going to be to start using  `var rand = new System.Random((int) _timing.CurTick.Value);` like how ClumsySystem does it, which guarantees we'll get the same randomness on both client and server side.

File: Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs [hunk 6]
```diff
@@ -136,7 +137,8 @@ private void OnTryChangePartDamage(Entity<BodyComponent> ent, ref TryChangePartD
                 // such as an animal, so we attack a random part.
                 if (args.Origin.HasValue)
                 {
-                    targetPart = GetRandomBodyPart(ent, targetEnt);
+                    // Evasion would trigger constantly if we don't target torso
+                    targetPart = args.CanEvade ? TargetBodyPart.Torso : GetRandomBodyPart(ent, targetEnt);
```
Comment (gluesniffler) [inline Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs:141]:
```suggestion
                    targetPart = GetRandomBodyPart(ent, targetEnt);

Comment (gluesniffler) [reply Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs:141]: No reason not to include it now since the prediction issue was solved.

Comment (Aviu00) [reply Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs:141]: This would cause mobs to try to hit head and fail.

Comment (gluesniffler) [reply Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs:141]: only 70% of the time, 1/6th of the time
gambling-loop

Comment (Aviu00) [reply Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs:141]: Yes, still annoying when you play as non-humanoid creature.

Comment (gluesniffler) [reply Content.Shared/_Shitmed/Body/Systems/SharedBodySystem.Targeting.cs:141]: weak. do as you please then

**Comment by JoeHammad1844** [comment]
hands and feet should be able to evade
wont change much but having just the head evade feels weird

**Review by Piras314 — APPROVED** [review]

**Pull request merged**